LLReactiveMatchers/Helpers/EXPExpect+LLReactiveMatchersExtensions.m

```diff
@@ -93,7 +93,7 @@ - (void) applyMatcherLLRMContinousAsync:(id<EXPMatcher>)matcher to:(NSObject *__
             }
             
             [[NSRunLoop currentRunLoop] runUntilDate:[NSDate dateWithTimeIntervalSinceNow:0.01]];
-            OSMemoryBarrier();
+            atomic_thread_fence(memory_order_seq_cst);
```

Comment:
Oy crap. Apparently, there's `OSATOMIC_USE_INLINED=1` define that removes deprecation and implements the old atomic api using the new one. 
This will solve you the iOS 10 problem, until we figure out what's the proper solution for all our code (ReactiveCocoa, Expecta, Specta, OCMock all use deprecated API)